### PR TITLE
Adds the option to perform Bulk exposure updates

### PR DESCRIPTION
**Pull request by MushuEE**
Fixes #
Part of the #243 effort

## Proposed Changes

* Add a way to perform bulk Exposure Key inserts. 
* This change decrease the time for the insert step of 1,400,000 Key publishes from ~18m to ~7s
* This decreased the overall test duration from ~20m to ~3m making it a candidate for presubmit. 
* I think this might also be an option for other tests but I am not familiar enough with the test to know if it will help.

```
{"severity":"INFO","timestamp":"2020-09-02T16:21:51.349986-07:00","caller":"database/connection.go:54","message":"Closing connection pool."}
--- PASS: TestExport (173.73s)
PASS
ok  	github.com/google/exposure-notifications-server/internal/performance	175.180s
```

/assign @chaodaiG 

*NOTE* While everything indicates that it is working. When I inspect the database (which I might have been doing wrong) I was not seeing rows being populated. (The counts are validated for inserts and the time does change for greater numbers of inserts). It could have been something else I was doing. @chaodaiG if you get a chance to sanity test this change it would be appreciated. I have been running with `./scripts/performance.sh --dev 100000`

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: internal/database/database_util.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ func NewTestDatabaseWithConfig(tb testing.TB) (*DB, *Config) {
 			"POSTGRES_USER=" + username,
 			"POSTGRES_PASSWORD=" + password,
 		},
+		// Mounts: []string{
```
Comment (MushuEE) [inline internal/database/database_util.go:79]: Remove, no longer needed

File: internal/performance/export_test.go [hunk 2]
```diff
@@ -141,25 +142,26 @@ func TestExport(t *testing.T) {
 	}
 
 	// Publish keys based on first batch of published keys
+	var revisedExposures []*publishmodel.Exposure
 	for i := 0; i < numPublishes; i++ {
 		if r := i % roughPerBatch; r == 0 { // increace start time after each batch
 			batchStartTime = batchStartTime.Add(exportPeriod)
 		}
-		var revisedExposures []*publishmodel.Exposure
+		// var revisedExposures []*publishmodel.Exposure
```
Comment (MushuEE) [inline internal/performance/export_test.go:150]: remove

File: internal/publish/database/exposure.go [hunk 3]
```diff
@@ -319,7 +319,6 @@ func prepareInsertExposure(ctx context.Context, tx pgx.Tx) (string, error) {
 				 created_at, local_provenance, sync_id, health_authority_id, report_type, days_since_symptom_onset)
 		VALUES
 			($1, $2, LOWER($3), $4, $5, $6, $7, $8, $9, $10, $11, $12, $13)
-		ON CONFLICT (exposure_key) DO NOTHING
```
Comment (mikehelmick) [inline internal/publish/database/exposure.go:322]: This line should not be removed.

File: internal/publish/database/exposure.go [hunk 4]
```diff
@@ -368,7 +407,7 @@ func executeReviseExposure(ctx context.Context, tx pgx.Tx, stmtName string, exp
 }
 
 // InsertAndReviseExposures transactionally revises and inserts a set of keys as necessary.
-func (db *PublishDB) InsertAndReviseExposures(ctx context.Context, incoming []*model.Exposure, token *pb.RevisionTokenData, tokenRequired bool) (int, error) {
+func (db *PublishDB) InsertAndReviseExposures(ctx context.Context, incoming []*model.Exposure, token *pb.RevisionTokenData, tokenRequired, bulkInsert bool) (int, error) {
```
Comment (mikehelmick) [inline internal/publish/database/exposure.go:410]: let's not put bulk insert in this same method.

bulk insert is meant for performance testing and shouldn't be in the main code path.

**Review by mikehelmick — CHANGES_REQUESTED** [review]

File: internal/publish/database/exposure.go (hunk 4 above)
Comment (MushuEE) [reply internal/publish/database/exposure.go:410]: Sounds good, I will break it out

Comment (MushuEE) [reply internal/publish/database/exposure.go:410]: @mikehelmick did you want it out of this package entirely or just broken out of `InsertAndReviseExposures` and make a new `BulkInsertExposures`

Comment (mikehelmick) [reply internal/publish/database/exposure.go:410]: can be in this package - but should be in a separate file, clearly marked as test infrastructure

**Comment by MushuEE** [comment]
@chaodaiG the naming feels a little wonky here but I wasn't sure what might be best.

**Comment by mikehelmick** [comment]
needs a rebase

**Comment by MushuEE** [comment]
Rebased, thanks

**Review by mikehelmick — APPROVED** [review]

**Pull request merged**